Theme.cfc

```diff
@@ -180,26 +181,23 @@ component{
 		if( pageSectionCatName != "none" ){
 			var oCategory = categoryService.findWhere( criteria={category=pageSectionCatName} );
 			var sPages = pageService.search( category=oCategory.getCategoryID(), sortOrder="order" );
-			
-			for ( var page in sPages.pages ) {
-				var sfields = page.getCustomFieldsAsStruct();
-					
-				for ( var field in aFieldKeys ) {
-			
-					// if page does not have field	
-					if( !structKeyExists( sfields, field  ) ){
-							
-						// create field
-						transaction{
-							var newCustomField = EntityNew( 'cbCustomField' );
-							newCustomField.setKey( field );
-							newCustomField.setValue( "" );
-							newCustomField.setRelatedContent( page );
+			transaction{
+				for ( var page in sPages.pages ) {
+					var sfields = page.getCustomFieldsAsStruct();
+						
+					for ( var field in aFieldKeys ) {
+				
+						// if page does not have field	
+						if( !structKeyExists( sfields, field  ) ){
 								
-							EntitySave( entity=newCustomField );
+							// create field
+								var args = { key = field, value = "" };
+								var oField = customFieldService.new(properties=args);
+								oField.setRelatedContent( page );
+								page.addCustomField( oField );	

```

Comment:
As per our testing, this is also working and is creating custom fields on the page.

This is the reference we followed to do this change.
https://github.com/Ortus-Solutions/ContentBox/blob/development/modules/contentbox/models/content/ContentService.cfc#L659

We could do what you have asked for but just wanted you to know why we did it like this.

Let us know.

Yogesh

Reply:
@lmajano Any comments?

Reply:
Ok, I get it.  Since you are adding the field to the page, which is in the hibernate session, and in the transaction block, the field is persisted. The problem is that its not intuitive. I would prefer that we remove the `page.addCustomField()` and add a `customFieldService.save( oField )` instead.  This denotes a saving transaction instead.  Once that is applied, then you can apply it to the other themes.